packages/kit/src/module/expose.ts

```diff
@@ -0,0 +1,44 @@
+import { resolve } from 'pathe'
+import { useNuxt } from '../context'
+import { addTemplate } from '../template'
+
+/**
+ * Expose virtual module config to the NuxtApp and nitro with TypeScript support.
+ */
+export function exposeRuntimeModuleConfig (moduleName: string, config: Record<string, any>) {
```

Comment:
I'm wondering if `config` could be typed to only support serializable values (i.e. a function is not) to prevent confusion/mistakes